### ISSUE TITLE
Next Step Microgrant: Operation Dragonborn

### ISSUE THREAD

**Issue by somya-15** [filecoin-project/devgrants#1232]
### 1. What is your project, and what problem does it solve? (max 100 words)
> ✨ In D&D, it is compulsory to create an environment that your character embodies, including an avatar.  This isn’t easy for first-timers or people with lack of time or imagination.  Our tools will help players with this.

“Operation Dragonborn” is the next step after our Top Quality Chainlink Hackathon project [D&D Diffusion](https://devpost.com/software/dndiffusion). Our sprint is focused on the narrow objective of training an AI art generator to understand the specific branded D&D term “Dragonborn.”

We want to create usable, unique, AI generated visual assets for a game that’s played by 50 million people a year.  We are going to bring people together to play games, inspire creativity and save time. Tabletop roleplaying is a multibillion dollar industry and not enough of it is decentralized. Our future platform will connect AI generated art to an NFT marketplace for human artists and allow us to participate in the ongoing conversation about art and media and the role of AI in art creation.

### 2. Project links
 Link to Github repo: [https://github.com/DnDnDiffusion](https://github.com/DnDnDiffusion)

Link to demo or website, if applicable: [https://devpost.com/software/dndiffusion](https://devpost.com/software/dndiffusion)

Link to our Notion (this grant application, but prettier): [Next-Steps-Grant-Proposal-Operation-Dragonborn](https://www.notion.so/Next-Steps-Grant-Proposal-Operation-Dragonborn-d5b28553577e42c6b016779de5f400fd)

Link to sprint progress: 

[Next Step Grant Progress](https://www.notion.so/Next-Step-Grant-Progress-d4e68e819f6346feaa62cf7304486548)

License: MIT
  
### 3. How is IPFS, Filecoin, or related technology used in this project? (max 200 words)
Right now in our MVPr, we used NFT.Storage to allow the user to programmatically save their AI generated avatar to decentralized storage, returning the CID after it’s stored to the front-end and enabling the minting of an NFT of that avatar.  

**The full picture**:

1. NextJS
2. PDF parser and data transformation (we will be deciding between our two versions, one JS and one Flask)
3. Stable-diffusion running in a cloud GPU, such as [runpod.io](http://runpod.io/)
4. A specially trained model to understand D&D Dragonborn and various classes.
5. NFT Smart Contract to allow users to save their avatars (and our allowing our front-end to display a [marketplace](https://imageai.app/explore/))

*Is this project contributing to the upcoming FVM launch? (Yes or **No**)* 

***NO***
  
### 4. How will you improve your project with this grant? What steps will you take to meet this objective? (max 200 words)
Imagine you’re playing D&D for the first time, as a Dragonborn Cleric, level one. What do you look like? Step one is to make a character sheet, then you’ll need to find an image or two for your character. Meanwhile, your whole party is waiting to start and saying interesting things about their characters!

During our hackathon, we created all the components required for our use-case but not a complete end-to-end deployment.  Operation Dragonborn is a series of sprints to integrate our existing components, and accurately visualize the content of a new player’s character sheet with our custom-trained stable diffusion models.  We will deploy a proof of concept for our test group to upload a [Fast Character](https://fastcharacter.com/) sheet PDF and visualize the race “[Dragonborn](https://5thsrd.org/character/races/dragonborn/)” from the D&D [Open License](https://5thsrd.org/). Users can save, share or mint it as an NFT.

### **End-to-end features we will develop in 1-2 sprints for this grant:**

- Functioning frontend
- Parsing PDF character sheets
- [NFT.Storage](http://nft.storage/) full NFT metadata
- Tested smart-contract
- Cloud GPU running stable diffusion

### 5. Do you agree to share grant reports upon request, including a final grant report at the end of the three month period?
Yes.  We are assembling a team of Technical Sponsors which include [Discordian](https://github.com/TheDiscordian/), [Dawn Kelly](https://github.com/dawnkelly09), [Developer Ally](https://github.com/DeveloperAlly), [Alexandre Nicastro](https://github.com/AlexxNica) and [Jenks Guo](https://github.com/jenksguo).  We intend for them to be our first test users, and have already created a group communication channel in the Filecoin Slack.
  
### 6. Does your proposal comply with our Community Code of Conduct?

Yes

### 7. Links and submissions

- For each team member(s), please list name, email, Github account, and role in the project.
    - Tippi Fifestarr: HR, Product & DevRel
        - [fifestarr21@gmail.com](mailto:fifestarr21@gmail.com)
        - [https://github.com/tippi-fifestarr](https://github.com/tippi-fifestarr)
    - Somya Tomar: Developer (R&D, full-stack, ML), Grant Liaison
        - [somyatomar15@gmail.com](mailto:somyatomar15@gmail.com)
        - [https://github.com/somya-15](https://github.com/somya-15)
    - Danjo the 0x: Chief Technology Officer
        - [danjo@danj0x.com](mailto:danjo@danj0x.com)
        - [https://github.com/danj-O](https://github.com/danj-O)
    - Jason Demma: Creative Director & Chief Humanity Officer
        - [creatjve@gmail.com](mailto:creatjve@gmail.com)
        - [jasondemma.com](http://jasondemma.com/)
    - Aire: Project Manager and Development Intern
        - [aire.forwork@gmail.com](mailto:aire.forwork@gmail.com)
        - [https://github.com/airecr](https://github.com/airecr)
    - Cromewar: Just a regular guy
        - [vasiliy@cromewar.com](mailto:vasiliy@cromewar.com)
        - [https://github.com/cromewar](https://github.com/cromewar)
- How did you learn about our microgrant program?
    - Tippi introduced our team to this grant.
- If your project was created as part of an event or hackathon:
    - Chainlink Fall 2022: [https://devpost.com/software/dndiffusion](https://devpost.com/software/dndiffusion)

### Additional questions:
What DnD race should we do next to train our AI model?

**Comment by ErinOCon** [comment]
Hi @somya-15, thank you for your proposal! This grant has been approved. We have sent an email to confirm payment details.

**Comment by tippi-fifestarr** [comment]
Thank you so much for this support, we will be starting our sprint tomorrow!

**Comment by tippi-fifestarr** [comment]
@ErinOCon We've met with Dawn and our Technical Sponsors, I think we are completed on this Grant? @dawnkelly09 @jenksguo

**Comment by ErinOCon** [comment]
Hi @tippi-fifestarr, thank you for being in touch! Can you provide a final report for this project [here](https://airtable.com/shrrtSlgdVvrPmAvr)? Please let us know if you have any questions!

**Comment by tippi-fifestarr** [comment]
> 

Just sent in a final report.  Cromewar is working on producing an end-to-end demo video with our current progress since our team is still updating the front-end with new designs and integrating last weeks user interview data (adding a reroll button).

**Comment by tippi-fifestarr** [comment]
> Hi @tippi-fifestarr, thank you for being in touch! Can you provide a final report for this project [here](https://airtable.com/shrrtSlgdVvrPmAvr)? Please let us know if you have any questions!

Hey did you get my response on the Airtable? 
This is the demo video we produced a [couple weeks ago and shared to our technical sponsors:](https://www.youtube.com/watch?v=DpK3QrO_cVo) 

Haven't heard back on https://github.com/filecoin-project/devgrants/issues/1377, and I know things are changing with the micro grants recently.  

We got started on integrating Bacalhau but have hit a lot of walls.  Getting that Grant would let us invest more developer time, for now we had to prioritize on getting a solid user experience instead of decentralizing our compute functions.
